packages/@aws-cdk/aws-eks/lib/cluster-resource-provider.ts

```diff
@@ -0,0 +1,52 @@
+import { NestedStack } from '@aws-cdk/aws-cloudformation';
+import * as iam from '@aws-cdk/aws-iam';
+import * as lambda from '@aws-cdk/aws-lambda';
+import { Construct, Duration, Stack } from '@aws-cdk/core';
+import * as cr from '@aws-cdk/custom-resources';
+import * as path from 'path';
+
+const HANDLER_DIR = path.join(__dirname, 'cluster-resource-handler');
+const HANDLER_RUNTIME = lambda.Runtime.NODEJS_12_X;
+
+export class ClusterResourceProvider extends NestedStack {
+  public static getOrCreate(scope: Construct) {
+    const stack = Stack.of(scope);
+    const uid = '@aws-cdk/aws-eks.ClusterResourceProvider';
+    return stack.node.tryFindChild(uid) as ClusterResourceProvider || new ClusterResourceProvider(stack, uid);
+  }
+
+  public readonly provider: cr.Provider;
+  public readonly roles: iam.IRole[];
+
+  constructor(scope: Construct, id: string) {
+    super(scope, id);
+
+    const onEvent = new lambda.Function(this, 'OnEventHandler', {
+      code: lambda.Code.fromAsset(HANDLER_DIR),
+      description: 'onEvent handler for EKS cluster resource provider',
+      runtime: HANDLER_RUNTIME,
+      handler: 'index.onEvent',
+      timeout: Duration.minutes(1)
+    });
+
+    const isComplete = new lambda.Function(this, 'IsCompleteHandler', {
+      code: lambda.Code.fromAsset(HANDLER_DIR),
+      description: 'isComplete handler for EKS cluster resource provider',
+      runtime: HANDLER_RUNTIME,
+      handler: 'index.isComplete',
+      timeout: Duration.minutes(1)
+    });
+
+    this.provider = new cr.Provider(this, 'Provider', {
+      onEventHandler: onEvent,
+      isCompleteHandler: isComplete,
+      totalTimeout: Duration.hours(1),
+      queryInterval: Duration.minutes(1)
```

Comment:
I would not find what the rate limits are for the EKS service endpoints. I feel like `1 minute` may be very aggressive... And could cause throttling if multiple clusters deploy simultaneously.

Reply:
Haven't seen any throttling issues, but I agree that 1 min is aggressive given in average it takes about 15 minutes to provision a cluster...

Reply:
Changed to 5min